### ISSUE TITLE
Regression: Fast route calculation failed

### ISSUE THREAD

**Issue by seahorse41** [osmandapp/OsmAnd#17156]
### Description
This worked fine in nightly version 4.4.0#35421m(4400) downloaded 2023-02-10
I noticed newer version so upgraded to 4.4.7-4407
Map rendering engine Version 2 (OpenGL)
Navigation totally fails.
### Steps to reproduce
Engine version 2.
Offline maps.
Press Navigation, select destination, get error message:
### Actual result
Fast route calculation failed
(Attempt to invoke virtual method 'net.osmand.binary.BinaryMapRouteReaderAdapter$RouteTypeRule net.osmand.binary.BinaryMapRouteReaderAdapter$RouteRegion.quickGetEncodingRule(int)' on a null object reference, fallback to slow calculation.

It keeps retrying and flashing the same message. I let it try for several minutes before giving up on it, closing 
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/14300304/88549874-b4f4-4b4b-844a-4f956b98b449"  width="300" height="auto"/> </th>
  </tr>
</table>  
and later changing the engine back to version 1. Route calculation works there.

Once home I changed engine to v2, closed it. Installed 4.4.0 and confirmed it worked as before. Closed the app. Then installed 4.4.7 again, then tried to duplicate what happened this morning. 
I could not duplicate it. I'm not sure what happened.

### Expected result
Route calculated, can press Start
### Your Environment
OsmAnd Version: 4.4.7 (4407)
Android/iOS version: 10.3.12
Device model: OnePlus 6T A6013

**Comment by Zirochkabila** [comment]
@seahorse41 In the latest nightly version and the release version, this cannot be repeated.
Please reinstall the app (back up the necessary data beforehand) and try again, or download the nightly version at https://osmand.net/docs/versions/nightly_versions
If the error persists, please send us your settings, profile - it may be the system settings that are causing the problem.

**Comment by vshcherb** [comment]
We need coordinates and the map to know to reproduce

**Comment by seahorse41** [comment]
Like I said, I could not reproduce it after reverting, reinstalling, and later upgrading to a newer version. Since this is the same device that was crashing with OpenGl-v2 engine a few months ago, I thought the report would be useful. 
I had a dentist appointment 1.5 hours away, and wanted to know my arrival time. I did not have time on the road to diagnose, restart, and try again.
Maybe it didn't have a gps lock soon enough when I started the navigation. It achieved lock before leaving. The screenshot is 4 minutes later, when it was clear it was not working.
Starting position: 48.75981 -118.60705
Destination: 48.54887 -117.90533
It always asks if I can access private roads, which is tiring. Maybe that was the null it didn't ask?
